### PR TITLE
Test and document set_user_data

### LINKED ISSUE #1397: Document set_user_data
The functionality to attach user data exists now but is undocumented. It's also a bit surprising that it can't be done at user creation.

### PR DESCRIPTION
**Pull request by eddyashton**
Fixes #1397.

- Adds `/app/log/private/admin_only` to the C++ logging app, using user-data as a very primitive ACL
- Expands documentation on `set_user_data` proposals
- Adds test case in `e2e_logging.py`

### CODE REVIEW

File: doc/members/open_network.rst [hunk 1]
```diff
@@ -69,7 +69,53 @@ The user can then make user RPCs, for example ``user_id`` to retrieve the unique
         "caller_id": 4
     }
 
-For each user CCF also stores arbitrary user-data in a JSON object, which can only be written to by members, subject to the standard proposal-vote governance mechanism. This lets members define initial metadata for certain users; for example to grant specific privileges, associate a human-readable name, or categorise the users. This user-data can then be read (but not written) by user-facing apps.
+User Data
+---------
+
+For each user, CCF also stores arbitrary user-data in a JSON object. This can only be written to by members, subject to the standard proposal-vote governance mechanism, via the ``set_user_data`` action. This lets members define initial metadata for certain users; for example to grant specific privileges, associate a human-readable name, or categorise the users. This user-data can then be read (but not written) by user-facing endpoints.
+
+For example, the C++ logging sample app has an endpoint (``/log/private/admin_only``) which uses user-data to restrict who is permitted to call it:
```
Comment (jumaffre) [inline doc/members/open_network.rst:77]: Grammar savvy: "For example, the `/log/private/admin_only` endpoint in the C++ sample app uses..." would IMO read better.

File: tests/e2e_logging.py [hunk 2]
```diff
@@ -388,8 +388,45 @@ def test_update_lua(network, args):
...
+    with primary.client(f"user{user_id}") as c:
+        r = c.post("/app/log/private/admin_only", {"id": 42, "msg": "hello world"})
+        assert r.status_code == http.HTTPStatus.FORBIDDEN.value, r.status_code
+
+    return network
+
+
 @reqs.description("Check for commit of every prior transaction")
-@reqs.supports_methods("/node/commit")
+@reqs.supports_methods("commit")
```
Comment (jumaffre) [inline tests/e2e_logging.py:429]: Should this requirement be removed as it is not application specific?

Comment (achamayou) [reply tests/e2e_logging.py:429]: We shouldn't list framework requirements.

Comment (eddyashton) [reply tests/e2e_logging.py:429]: Yeah this is fair, probably not worth requiring anything that's available-by-default. I'll remove

**Pull request merged**